src/prefect/transactions.py

```diff
@@ -104,6 +104,7 @@ def __enter__(self):
             # either inherit from parent or set a default of eager
             if parent:
                 self.commit_mode = parent.commit_mode
+                self._stored_values = parent._stored_values.copy()
```

Comment:
With this implementation, child transactions could modify values handled via reference, like the contents of a nested dictionary. Is that acceptable?

Reply:
Hmm good call - no I don't think that's acceptable; I suppose I need to do a deepcopy!

Reply:
Updated to deepcopy